### PR TITLE
Extend trigger type

### PR DESCRIPTION
**Pull request by itai-codefresh**

### CODE REVIEW

File: pkg/backend/dev_compose_types.json [hunk 1]
```diff
@@ -18,15 +18,31 @@
...
+                "push"
             ]
         },
         {
             "type": "cron",
             "kind": "codefresh",
             "service-url": "http://cronus:10002",
             "uri-template": "cron:codefresh:{{expression}}:{{message}}",
-            "uri-regex": "^cron:codefresh:@yearly|@annually|@monthly|@weekly|@daily|@midnight|@hourly|([a-zA-Z0-9*/,-?]+\\s?){6}:[a-zA-Z0-9_-]{2,64}$",
+            "uri-regex": "^Zcron:codefresh:@yearly|@annually|@monthly|@weekly|@daily|@midnight|@hourly|([a-zA-0-9*/,-?]+\\s?){6}:[a-zA-Z0-9_-]{2,64}$",
```
Comment (alexei-led) [inline pkg/backend/dev_compose_types.json]: `Z` in wrong place - somehow moved from `A-Z0-9` to the string beginning

File: pkg/backend/test_types.json [hunk 2]
```diff
@@ -36,7 +52,23 @@
                     "type": "string",
                     "validator": "^[a-z0-9_-]+$",
                     "required": true
+                },
+                {
```
Comment (alexei-led) [inline pkg/backend/test_types.json]: remove `action` from `config` we have it as separate section

File: pkg/codefresh/pipeline.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ type (
 
 var (
 	// RequestID request ID for logging
-	RequestID = "X-Request-ID"
+	RequestID = "X-Request-Id"
```
Comment (alexei-led) [inline pkg/codefresh/pipeline.go]: Is it supposed to be `Id` not `ID`?

File: pkg/backend/test_types.json [hunk 4]
```diff
@@ -16,14 +16,30 @@
...
```
Comment (alexei-led) [inline pkg/backend/test_types.json]: remove `action` from `config` we have it as separate section

File: pkg/backend/dev_external_types.json [hunk 5]
```diff
@@ -18,15 +18,31 @@
                     "type": "string",
                     "validator": "^[a-z0-9_-]+$",
                     "required": true
+                },
+                {
```
Comment (alexei-led) [inline pkg/backend/dev_external_types.json]: remove `action` from `config` we have it as separate section

File: pkg/backend/dev_compose_types.json [hunk 6]
```diff
@@ -18,15 +18,31 @@
                     "type": "string",
                     "validator": "^[a-z0-9_-]+$",
                     "required": true
+                },
+                {
```
Comment (alexei-led) [inline pkg/backend/dev_compose_types.json]: remove `action` from `config` we have it as separate section

File: hermes/templates/configmap.yaml [hunk 7]
```diff
@@ -30,7 +30,23 @@ data:
                 "type": "string",
                 "validator": "^[a-z0-9_-]+$",
                 "required": true
+              },
+              {
```
Comment (alexei-led) [inline hermes/templates/configmap.yaml]: remove `action` from `config` we have it as separate section

File: pkg/codefresh/pipeline.go (hunk 3 above)
Comment (itai-codefresh) [reply pkg/codefresh/pipeline.go]: it is supposed to be ID, but for some reason go transforms it to Id, so it didn't work so i changed it

**Pull request merged**